dart/lib/src/sentry_attachment/sentry_attachment.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+// https://develop.sentry.dev/sdk/features/#attachments
+// https://develop.sentry.dev/sdk/envelopes/#attachment
+
+typedef ContentLoader = FutureOr<Uint8List> Function();
+
+/// Arbitrary content which gets attached to an event.
+class SentryAttachment {
+  SentryAttachment.fromLoader({
+    required ContentLoader loader,
+    required this.filename,
+    String? attachmentType,
+    this.contentType,
+  })  : _loader = loader,
+        attachmentType = attachmentType ?? AttachmentType.attachment;
+
+  /// Creates an [SentryAttachment] from a [Uint8List]
+  SentryAttachment.fromUint8List(
+    Uint8List bytes,
+    String fileName, {
+    String? contentType,
+    String? attachmentType,
+  }) : this.fromLoader(
+          attachmentType: attachmentType,
+          loader: () => bytes,
+          filename: fileName,
+          contentType: contentType,
+        );
+
+  /// Creates an [SentryAttachment] from a [List<int>]
+  SentryAttachment.fromIntList(
+    List<int> bytes,
+    String fileName, {
+    String? contentType,
+    String? attachmentType,
+  }) : this.fromLoader(
+          attachmentType: attachmentType,
+          loader: () => Uint8List.fromList(bytes),
+          filename: fileName,
+          contentType: contentType,
+        );
+
+  /// Creates an [SentryAttachment] from [ByteData]
+  SentryAttachment.fromByteData(
+    ByteData bytes,
+    String fileName, {
+    String? contentType,
+    String? attachmentType,
+  }) : this.fromLoader(
+          attachmentType: attachmentType,
+          loader: () => bytes.buffer.asUint8List(),
+          filename: fileName,
+          contentType: contentType,
+        );
+
+  /// Attachment type.
+  /// Should be one of types given in [AttachmentType].
+  final String attachmentType;
+
+  /// Attachment content.
+  /// Is loaded while sending this attachment.
+  FutureOr<Uint8List> get bytes => _loader();
+
+  final ContentLoader _loader;
+
+  /// Attachment file name.
+  final String filename;
+
+  /// Attachment content type.
+  /// Inferred by Sentry if it's not given.
+  final String? contentType;
+}
+
+/// Attachment type.
+class AttachmentType {
```

Comment:
should it be a different file?

Reply:
I'm not sure either. Maybe they should be fields of the attachment itself?

Reply:
on Java we have as a property in the Attachment class called `DEFAULT_ATTACHMENT_TYPE`, which defaults to `event.attachment`, we don't have the concept of `AttachmentType` unless someone else passes a custom string.
so maybe those constants should be part of the SentryAttachment itself